OpenSkyPlusApi/DeviceControls.cs

```diff
@@ -1,230 +1,380 @@
-                .GetProperty("MPGEGBMCFFN", BindingFlags.Static | BindingFlags.Public)?
-                .GetValue(null).ToString();
-
-            if (launchMonitorAppVersion != null)
-                _logger.LogDebug("Assembly-CSharp.dll is loaded");
-            else
-                Terminate("Could not find Assembly-CSharp.dll. Check the app installation and try again.");
-
-            if (launchMonitorAppVersion != OpenSkyPlusApi.SupportedVersion)
-            {
-                _logger.LogWarning(
-                    $"{_config.AppSettings.LaunchMonitor} application version does not match supported version. Application may not work correctly.");
-                _logger.LogWarning($"Supported version: {OpenSkyPlusApi.SupportedVersion}");
-                _logger.LogWarning($"Found: {launchMonitorAppVersion}");
-            }
-        }
-        catch
-        {
-            Terminate($"{_config.AppSettings.LaunchMonitor} assemblies not found. Check the app path and try again.");
-        }
-
-        // THIS LINE CHECKS TO MAKE SURE THE LICENSE IS VALID. DO NOT DELETE THIS LINE OR THE CHECK WILL NOT WORK!
-        if (!CheckLicense()) Terminate("Could not verify app license");
-
-        try
-        {
-            InitializeLaunchMonitorWrapper();
-            InitializeLaunchMonitorController();
-        }
-        catch (Exception ex)
-        {
-            _logger.LogFatal($"Could not initialize {_config.AppSettings.LaunchMonitor} wrapper functions:\n{ex}");
-            Terminate($"Failed to locate {_config.AppSettings.LaunchMonitor} monitor wrapper functions.\n" +
-                      $"Is this Api compatible with the version {_config.AppSettings.LaunchMonitor}?");
-        }
-
-        return;
-
-        static bool CheckLicense()
-        {
-            try
-            {
-                var licenseStatusProperty = AccessTools
-                    .TypeByName($"{_config.AppSettings.LaunchMonitor}Wrapper.EONIOHDNGND+MOBEJGCNPEM")
-                    .GetProperty("GNFAINMEEHO");
-
-                var licenseStatusInstance = AccessTools
-                    .TypeByName($"{_config.AppSettings.LaunchMonitor}Wrapper.EONIOHDNGND+IBHCDNLENEF")?
-                    .GetProperty("GEDJNGPNLKO")?.GetValue(AccessTools.TypeByName("Security.ELOOGCNONDM")?
-                        .GetProperty("IBHCDNLENEF")?.GetValue(AccessTools.TypeByName("Security.AOMKMFBGCGG")?
-                            .GetProperty("KGJFDOMIPBN", BindingFlags.Public | BindingFlags.Static)?
-                            .GetValue(null)));
-
-                if ((int?)licenseStatusProperty?.GetValue(licenseStatusInstance) == 0)
-                {
-                    _logger.LogInfo("Play and Improve license is valid");
-                    return true;
-                }
-            }
-            catch (Exception ex)
-            {
-                _logger.LogDebug($"Exception during license check:\n{ex}");
-                return false;

```

Comment:
What's this removed for?

Reply:
Probably a resharper comment that it is not necessary ;)

Reply:
My comment was just a statement, no problem for me to put it back.